salt/utils/win_pwsh.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+import salt.modules.cmdmod
+import salt.utils.json
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+
+__salt__ = {"cmd.run_all": salt.modules.cmdmod.run_all}
+log = logging.getLogger(__name__)
+
+__virtualname__ = "win_pwsh"
+
+
+def __virtual__():
+    """
+    Only load if windows
+    """
+    if not salt.utils.platform.is_windows():
+        return False, "This utility will only run on Windows"
+
+    return __virtualname__
+
+
+def run_dict(cmd, cwd=None):
+    """
+    Execute the powershell command and return the data as a dictionary
+
+    Args:
+
+        cmd (str): The powershell command to run
+
+        cwd (str): The current working directory
+
+    Returns:
+        dict: A dictionary containing the output of the powershell command
+
+    Raises:
+        CommandExecutionError:
+            If an error is encountered or the command does not complete
+            successfully
+    """
+    if "convertto-json" not in cmd.lower():
+        cmd = f"{cmd} | ConvertTo-Json"
+    if "progresspreference" not in cmd.lower():
+        cmd = f"$ProgressPreference = 'SilentlyContinue'; {cmd}"
+    log.debug("PowerShell: %s", cmd)
```

Comment:
Could this possibly log sensitive information?

Reply:
I removed the log message. I believe ``cmd.run_all`` still logs the command being run though, so I'm not sure this fixes the issue.